edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -95,30 +96,106 @@ impl std::fmt::Debug for DockerModuleRuntime {
-                })?;
-                Ok(base64::encode_config(&json, base64::URL_SAFE))
-            },
-        );
+        let image_with_tag = config.image().to_string();
+        // check if the serveraddress exists & check if it exists in notary_registries
+        let registry_auth = config.auth();
+        let (registry_hostname, registry_username, registry_password) = match registry_auth {
+            Some(a) => (a.serveraddress(), a.username(), a.password()),
+            None => (None, None, None),
+        };
+        let hostname = registry_hostname.unwrap_or_default();
+        let username = registry_username.unwrap_or_default();
+        let password = registry_password.unwrap_or_default();
+
+        let image = if !hostname.is_empty() && !username.is_empty() && !password.is_empty() {
+            let config_path = self.notary_registries.get(hostname);
+            if let Some(c) = config_path {
+                info!("{} is enabled for notary content trust", hostname);
+                let config_path_buf = c;
+                let config_path_string_result =
+                    config_path_buf.clone().into_os_string().into_string();
+                let config_path_str;
+                match config_path_string_result {
+                    Ok(c) => config_path_str = c,
+                    Err(_) => config_path_str = "".to_string(),
+                }
+                let mut image_with_tag_parts = image_with_tag.split(':');
+                let gun = image_with_tag_parts
+                    .next()
+                    .expect("split always returns atleast one element")
+                    .to_owned();
+                let gun: Arc<str> = gun.into();
+                let tag = image_with_tag_parts.next().unwrap_or("latest").to_owned();
+                let notary_auth = format!("{}:{}", username, password);
+                let notary_auth = base64::encode(&notary_auth);
+                let lock = self.notary_lock.clone();
+                let mutex = MutexFuture(lock);
+                future::Either::A(
+                    mutex
+                        .and_then({
+                            let gun = gun.clone();
+                            move |lock| {
+                                notary::notary_lookup(
+                                    &notary_auth,
+                                    &gun,
+                                    &tag,
+                                    &config_path_str,
+                                    lock,
+                                )
+                            }
+                        })
+                        .map(move |digest| (format!("{}@{}", gun, digest), true)),
+                )
+            } else {
+                info!("{} is not enabled for notary content trust", hostname);
+                future::Either::B(futures::future::ok((image_with_tag, false)))
+            }
+        } else {
```

Comment:
I am curious what could result in this path?

Reply:
Else flow in line 171 results when there is no no auth creds for a registry server. it could be a public server like mcr or docker hub then image gets pulled by tag